### ISSUE TITLE
`no-unused-vars` false positive on increment

### ISSUE THREAD

**Issue by mymoomin** [oxc-project/oxc#5671]
Using a variable to look up a key in an object then `++`ing it doesn't count as a use, but `+= 1` does. Eslint does not report anything in either case

```js
const counts = {};

for (const char of "hello") {
  counts[char]++;
}

for (const char of "world") {
  counts[char] += 1;
}
```
gets
```
> npx oxlint@latest
  ⚠ eslint(no-unused-vars): Variable 'char' is declared but never used.
   ╭─[testing.js:3:12]
 2 │ 
 3 │ for (const char of "hello") {
   ·            ──┬─
   ·              ╰── 'char' is declared here
 4 │   counts[char]++;
   ╰────
  help: Consider removing this declaration.
```

**Comment by mymoomin** [comment]
The same thing happens for arrays
```js
const counts = [0, 0, 0, 0, 0, 0];

for (const index of [0, 1, 2]) {
  counts[index]++;
}

for (const char of [3, 4, 5]) {
  counts[char] += 1;
}

console.log(counts);
```

```
> npx oxlint@latest

  ⚠ eslint(no-unused-vars): Variable 'index' is declared but never used.
   ╭─[testing.js:2:12]
 1 │ const counts = [0, 0, 0, 0, 0, 0];
 2 │ for (const index of [0, 1, 2]) {
   ·            ──┬──
   ·              ╰── 'index' is declared here
 3 │   counts[index]++;
   ╰────
  help: Consider removing this declaration.
```
